src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncExceptionHandlerRewriter.cs

```diff
@@ -516,7 +516,7 @@ private BoundNode VisitCatchBlock(BoundCatchBlock node, AwaitCatchFrame parentAw
             if (!_analysis.CatchContainsAwait(node))
             {
                 var origCurrentAwaitCatchFrame = _currentAwaitCatchFrame;
-                _currentAwaitCatchFrame = null;
+                _currentAwaitCatchFrame = parentAwaitCatchFrame;
```

Comment:
>_currentAwaitCatchFrame = parentAwaitCatchFrame [](http://example.com/codeflow?start=16&length=47)

Are we testing the case where `_currentAwaitCatchFrame` and `parentAwaitCatchFrame` are both not `null` entering this method? Is that case possible or interesting?

Reply:
I think that would happen when there are multiple sibling `catch` clauses (then `_currentAwaitCatchFrame` corresponds to the previous sibling `catch` clause as explained in https://github.com/dotnet/roslyn/pull/70513#discussion_r1371475860) nested in another `catch` clause (then `parentAwaitCatchFrame` corresponds to the parent `catch` clause).

I don't think it's interesting since we just want to always ignore the `_currentAwaitCatchFrame` here as we did previously - because it corresponds to the previous sibling `catch` clause which is not useful to us.